examples/images_contours_and_fields/pcolormesh_grids.py

```diff
@@ -0,0 +1,132 @@
+"""
+============================
+Pcolormesh grids and shading
+============================
+
+`.axes.Axes.pcolormesh` and `~.axes.Axes.pcolor` have a few options for
+how grids are laid out and the shading between the grid points.
+
+Generally, if *Z* is shape *(M, N)* then the grid *X* and *Y* can be
+specified with either shape *(M+1, N+1)* or *(M, N)*, depending on the
+argument for the ``shading`` keyword argument.  Note that below we specify
+vectors *x* as either length N or N+1 and *y* as length M or M+1, and
+`~.axes.Axes.pcolormesh` internally makes the mesh matrices *X* and *Y* from
+the input vectors.
+
+"""
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+
+###############################################################################
+# Flat Shading
+# ------------
+#
+# The grid specification with the least assumptions is ``shading='flat'``
+# and if the grid is one larger than the data in each dimesion, i.e. has shape
+# *(M+1, N+1)*.  In that case *X* and *Y* sepcify the corners of quadrilaterals
+# that are colored with the values in *Z*. Here we specify the edges of the
+# *(3, 5)* quadrilaterals with *X* and *Y* that  are *(4, 6)*.
+
+np.random.seed(19680801)
+Z = np.arange(15).reshape(3, 5)
+x = np.arange(6)  # len = 6
```

Comment:
`len = 6` (also below a few times) is a bit mysterious here, perhaps `x = np.arange(Z.shape[1] + 1)` may show intent more clearly?

Reply:
Changed to `nrows = 3` etc and use `nrows + 1` where appropriate...